TypeCobol/Compiler/Scanner/MultilineScanStateSnapshot.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TypeCobol.Compiler.Scanner
+{
+    /// <summary>
+    /// Snapshot of a  MultilineScanState
+    /// </summary>
+    public class MultilineScanStateSnapshot : MultilineScanState
```

Comment:
So basically `MultilineScanStateSnapshot` is a readonly `MultilineScanState`. I don't think we need this new class, you can set the scan state of a partial cobol word with a clone of the current relevant scan state. this is more or less what we are already doing with the `InitialScanState` property in `TokensLine`.

Reply:
I will try a clone, I use property values because most of them are calculated properties.